docs/design/principles.md

```diff
@@ -1,54 +1,47 @@
 # ICU4X Design Principles & Decisions
 
-## Guiding principles
-
 These principles are not cast in stone, but are strong guidelines for developers.
 
-### Immutable objects
+## i18n Best Practices
+
+Above all, ICU4X must provide modern, standards-compliant APIs that encourage best i18n practices and produce correct results for all languages and locales. No language or locale should be at a structural disadvantage.
+
+## Code Style
 
-No object is to be mutated after creation.
+All ICU4X code must conform to the [style guide](../process/style_guide.md), including the following:
 
-### No I/O in the core library
+### Safety
 
-Leave I/O to the wrapper or clients. Pass data as reference to the methods.
+All standard Rust practices regarding mutability, lifetimes and safety must be followed.
 
-***
-*NOTE*: Clients can mmap, async fetch, package data with the library...
-***
+### No standard library dependencies in the core library
+
+The core icu4x library (the `icu` and `icu_capi` crates and all of their direct and indirect dependencies, not including dev-dependencies) should be `#[no_std]`, but may use the `alloc` crate.
```

Comment:
issue: this needs to be defined in a way that excludes the data provider impls

Reply:
The data provider impls are not dependencies of `icu`.  In `icu_capi` they are dependencies only insofar as they are behind flags.  Is that what you meant?

Reply:
Yeah

Reply:
ok I removed the reference to `icu_capi` and just left `icu`